packages/analytics/analytics-chart/src/components/chart-plugins/ChartLegend.vue

```diff
@@ -54,8 +57,19 @@ const props = defineProps({
     default: () => null,
   },
 })
+const legendContainerRef = ref<HTMLElement>()
+const shouldTruncate = ref(false)
+
+const checkForWrap = () => {
+  if (legendContainerRef.value) {
+    const scrollHeight = legendContainerRef.value.scrollHeight
+    if (scrollHeight > parseInt(KUI_SPACE_150, 10)) {
+      shouldTruncate.value = true
+    }
+  }
+}
 
-const itemsRef = toRef(props, 'items')
+watch(() => props.items, checkForWrap, { immediate: true })
```

Comment:
I think this should have a `flush: 'post`` -- experimenting with the sandbox, it doesn't actually trigger wrapping the instant that a dataset starts to trigger scrolling.  At the time this `watch` runs, the DOM hasn't updated yet.

Reply:
that sounds good